synapse/config/tls.py

```diff
@@ -81,6 +81,28 @@ def read_config(self, config, config_dir_path, **kwargs):
             "federation_verify_certificates", True
         )
 
+        # Minimum TLS version to use for outbound federation traffic
+        self.federation_minimum_tls_client_version = self.parse_float(
```

Comment:
TBH I'd rather parse these as strings than worry about approximate float stuff. We don't seem to actually use the fact they're floats?

Reply:
yeah switched to making it a string